src/main/java/org/commonjava/util/partyline/JoinableFile.java

```diff
@@ -468,7 +468,7 @@ public int read()
             read++;
 
 //            logger.trace( "Joint: {} Read count: {}, returning: {}", jointIdx, read, Integer.toHexString( result ) );
-            return result;
+            return result < 0 ? result + 256 : result;
```

Comment:
I don't like this solution very much because it is some counting magic :-) How about changing this to
`return result & 0xff`
?